### ISSUE TITLE
[suggestion] Have docker build sniff the Dockerfile for possible simple errors before starting

### ISSUE THREAD

**Issue by smolinari** [docker/docker#22759]
Hi,

I am new to Docker, so maybe this is already possible or I am just doing something wrong (well, yeah, my inexperience is causing the issue.... LOL!)

The problem. 

If the dockerfile is incorrectly formatted, `docker build` runs through its motions up until the point it doesn't know what to do (where the improper formatting is) then it fails building the Dockerfile. In other words, I might be building PHP from scratch and at some point 20 minutes later, because a `\`is forgotten at the end of a line, the `docker build`errors saying it doesn't know what "&&" is and the build process is killed. 

The possible solution.  

Have `docker build` first read through the file for possible and simple "stupid" mistakes and stop with a warning about the issue, before the build process starts. 

I think that would save a lot of people, especially beginners like me, a lot of time.

Scott

**Comment by thaJeztah** [comment]
Thanks for the suggestion; I think this is a duplicate of  https://github.com/docker/docker/issues/19644 (and https://github.com/docker/docker/issues/19593) it may not be simple to detect these errors, because the actual *code* that executes inside `RUN` lines, isn't parsed by the builder, but sent to the shell inside the container as-is.

I'll close this issue as a duplicate, but thanks again for suggesting

**Comment by smolinari** [comment]
Thanks for the reply. Sorry for making the duplicate. Does that mean checking for general issues in the Dockerfile format isn't possible at all? I don't mean the commands inside the `RUN` lines, but just the overall composition of the file, like the Docker commands and the missing "\" for end of lines etc.

Scott

**Comment by thaJeztah** [comment]
It could be possible, although we haven't made that decision yet; not sure if that's something needed as part of the core functionality of Docker, given that it could also be implemented as an external tool

**Comment by smolinari** [comment]
If checking the file format can save any Docker user's time, it should be in the core. 

Scott
